### PR TITLE
feat: handle fixup and squash messages

### PR DESCRIPTION
**Pull request by tomassebestik**
This PR handles the usage of the `git commit --fixup` command.

Fixup and squash messages are ignored by commit linter as they are temporary and will not remain in the git history.

## Related
- Closes RDT-537

### CODE REVIEW

**Comment by tomassebestik** [comment]
FYI @KonstantinKondrashov : That should solve your request described in internal JIRA RDT-537

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# The global '*' pattern specifies code owners for all files in the repository.
+*       @tomassebestik @kumekay
```
Comment (kumekay) [inline .github/CODEOWNERS:2]: I would suggest adding more people in different locations. 

An even better approach is to create a team in the espressif organization and add people there, let's discuss it 1on1

**Review by kumekay — COMMENTED** [review]
@tomassebestik  Thank you for the PR, the solution is ok, but how can we be sure that these fixup commits will always be squashed before the merge?

File: .github/CODEOWNERS (hunk 1 above)
Comment (tomassebestik) [reply .github/CODEOWNERS:2]: I'm still thinking about how to set this up in a meaningful and useful way. We need to set some code owners here to use the branch protection rule `Require review from Code Owners`, which seems to me to be a useful and logical rule. 

In simple projects, it probably doesn't make much sense to add more granularity. Usually, just one or two people are responsible for the entire repository, including docs and CI.


An individual discussion would be appreciated. Thanks!

**Comment by tomassebestik** [comment]
> @tomassebestik Thank you for the PR, the solution is ok, but how can we be sure that these fixup commits will always be squashed before the merge?

I understand your concern, but this is probably beyond what this tool is capable of. Usually, a pre-commit linter is used along with Danger (GL or GH), and cases like unsquashed fixup commits will be caught by Danger.

I don't think it would be very wise to try implement additional logic for that in the pre-commit hook for the commit linter since we probably want to keep this tool simple and straightforward. WDYT?

**Pull request merged**